### ISSUE TITLE
Bug: `no-duplicate-imports` is not catching `import {wow, wow} from "something"`

### ISSUE THREAD

**Issue by pvinis** [eslint/eslint#16680]
### Environment

Node version: v16.10.0
npm version: v7.24.0
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: darwin 22.2.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
no-duplicate-imports: "error"
```
</details>

```js
import { wow, wow } from "something"
```

### What did you expect to happen?

I expected the second `wow` to be removed, autofixed.

### What actually happened?

Nothing.

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

I made a tiny silly PR here, to add the test case that should be failing. Feel free to take over/help me/close. https://github.com/eslint/eslint/pull/16679

**Comment by mdjermanovic** [comment]
Hi @pvinis, thanks for the issue!

`no-duplicate-imports` doesn't catch this case because it's a parsing error and will already be reported by the parser:

> Parsing error: Identifier 'wow' has already been declared

[Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWR1cGxpY2F0ZS1pbXBvcnRzOiBbXCJlcnJvclwiXSAqL1xuXG5pbXBvcnQgeyB3b3csIHdvdyB9IGZyb20gXCJzb21ldGhpbmdcIiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnsiZXM2Ijp0cnVlfX19)

**Comment by mdjermanovic** [comment]
Also, note that this rule is about disallowing multiple `import` statements from the same module. It doesn't check whether the same name is imported multiple times. For example, `import { wow as a, wow as b } from "something"` is valid code for this rule.

https://eslint.org/docs/latest/rules/no-duplicate-imports

> This rule requires that all imports from a single module that can be merged exist in a single `import` statement

**Comment by pvinis** [comment]
hm.. but shouldnt the wow,wow case be fixable? to make it just one wow? 🤔

**Comment by mdjermanovic** [comment]
ESLint doesn't autofix parsing errors. Fixes are generated by rules, and rules are run only after successful parsing.
